include/ginkgo/core/base/executor.hpp

```diff
@@ -900,7 +900,7 @@ class Executor : public log::EnableLogging<Executor> {
      *
      * @param ptr  pointer to the allocated memory block
      */
-    virtual void raw_free(void* ptr) const noexcept = 0;
+    virtual void raw_free(const void* ptr) const noexcept = 0;
```

Comment:
I guess technically interface breaking ? But we can probably ignore it if expect no outside application to have derived from the `Executor` class yet ?

Reply:
Now that I think about this again, I believe maybe a more appropriate solution would be to solve this mismatch on the `deleter` or `Array<const ...>` level, maybe with a special deleter for `const` data that casts `const`-ness away?

Reply:
Yes, I think if that works, that might be better solution.

Reply:
I think this much interface break should be okay. We should just document this in the release notes whenever this is released. But I agree that dealing with this in Array's internal deleter might be a bit better.